src/pages/static/js/settings.js

```diff
@@ -8,6 +8,108 @@
 let settingsAPI;
 let eventCleanupFunctions = [];
 
+// Hosts for built-in engines (lowercase)
+const BUILTIN_SEARCH_HOSTS = new Set([
+  "duckduckgo.com",
+  "search.brave.com",
+  "www.ecosia.org",
+  "kagi.com",
+  "www.startpage.com",
+]);
+
+function isBuiltInSearchTemplate(tpl) {
+  try {
+    // Replace %s with a sample to parse a real URL
+    const url = new URL(tpl.replace("%s", encodeURIComponent("test")));
+    const host = url.hostname.toLowerCase();
+
+    if (!BUILTIN_SEARCH_HOSTS.has(host)) return false;
+
+    // Extra safety: common built-in paths that indicate "regular web search"
+    const p = url.pathname.toLowerCase();
+
+    // duckduckgo: /  (query in ?q=)
+    // brave: /search
+    // ecosia: /search
+    // kagi: /search
+    // startpage: /do/search
+    if (
+      host === "duckduckgo.com" || // homepage with ?q=
+      (host === "search.brave.com" && p.includes("/search")) ||
+      (host === "www.ecosia.org" && p.includes("/search")) ||
+      (host === "kagi.com" && p.includes("/search")) ||
+      (host === "www.startpage.com" && p.includes("/do/search"))
+    ) {
+      return true;
+    }
+    return false;
+  } catch {
+    return false;
+  }
+}
+
+// Validates a custom search template:
+// - must start with http:// or https://
+// - must include %s
+// - must be a parsable URL once %s is replaced
+function validateSearchTemplate(tpl) {
+  if (typeof tpl !== "string")
+    return { valid: false, reason: "Template must be a string." };
+
+  const s = tpl.trim();
+  if (!s) return { valid: false, reason: "Template cannot be empty." };
+  if (!s.startsWith("http://") && !s.startsWith("https://")) {
+    return {
+      valid: false,
+      reason: "Template must start with http:// or https://.",
```

Comment:
It's a peer-to-peer web browser. We might support search engines (like [ipfs-search](https://github.com/ipfs-search/ipfs-search)
) in the future that can be accessed via ipns:// or hyper://.

So please remove this condition.